zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -827,9 +902,28 @@ where
                     .as_ref()
                     .map_or_else(MemoBytes::empty, |m| m.clone());
 
+                #[cfg(feature = "orchard")]
+                if let Some(orchard_receiver) = ua.orchard() {
```

Comment:
Bug: this (and the code below) is not using the `payment_pools` metadata now-encoded in the proposal, and is creating UA outputs to the preferred pool of this wallet version (instead of the wallet version that made the proposal).